### PR TITLE
Objective refactor

### PR DESCRIPTION
**Pull request by AnturK**
Objectives are no longer stored on mind only antagonists or teams.
Also various cleanups.
That's all. Next step is removal of explicit owner so they're stateless but that's another PR.

### CODE REVIEW

File: code/datums/mind.dm [hunk 1]
```diff
@@ -449,157 +438,92 @@
...
 						if (QDELETED(target))
 							return
 						else if(target == "(new custom antag)")
 							target_antag = add_antag_datum(/datum/antagonist/custom)
 						else
 							target_antag = target
 
-		var/new_obj_type = input("Select objective type:", "Objective type", def_value) as null|anything in list("assassinate", "maroon", "debrain", "protect", "destroy", "prevent", "hijack", "escape", "survive", "martyr", "steal", "download", "nuclear", "capture", "absorb", "custom")
-		if (!new_obj_type)
+		var/list/allowed_types = list(
```
Comment (Jared-Fogle) [inline code/datums/mind.dm]: typecacheof(/datum/objective, TRUE), then use typecache everywhere else?

Comment (AnturK) [reply code/datums/mind.dm]: This is not all objective types, only the ones that support admin editing.

File: code/__HELPERS/roundend.dm [hunk 2]
```diff
@@ -510,10 +510,12 @@
...
 
 
-/proc/printobjectives(datum/mind/ply)
+/proc/printobjectives(list/objectives)
+	if(!objectives || !objectives.len)
+		return
 	var/list/objective_parts = list()
 	var/count = 1
-	for(var/datum/objective/objective in ply.objectives)
+	for(var/datum/objective/objective in objectives)
```
Comment (ExcessiveUseOfCobblestone) [inline code/__HELPERS/roundend.dm]: Should this be typeless? You should only have objectives in the objective list.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/mind.dm [hunk 3]
```diff
@@ -449,157 +438,92 @@
...
+			/datum/objective/martyr,
+			/datum/objective/steal,
+			/datum/objective/download,
+			/datum/objective/nuclear,
+			/datum/objective/capture,
+			/datum/objective/absorb,
+			/datum/objective/custom
+		)
+
+		var/choices = list()
```
[the remark this answers is not in the document]
Comment (AnturK) [reply code/datums/mind.dm]: Yeah, doesn't really need to be done more than once.

**Pull request merged**